### PR TITLE
heroku ci:debug

### PR DESCRIPTION
**Pull request by appleton**
Add a command to start a test run and open a bash session to debug it.

### CODE REVIEW

File: commands/ci/debug.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    dyno.start()
+  } catch (err) {
+    if (err.exitCode) cli.exit(err.exitCode, err)
+    else throw err
+  }
+
+  // TODO: uncomment when this endpoint is deployed
+  // yield cli.action(
+  //   'Cancelling test run',
+  //   api.updateTestRun(heroku, testRun.id, { status: 'cancelled' })
```
Comment (joshwlewis) [inline commands/ci/debug.js]: This endpoint currently requires a message as well.

File: commands/ci/debug.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    'no-tty': context.flags['no-tty'],
+    attach: true,
+    env: 'HEROKU_SUPPRESS_LOGGING=true',
+    size: context.flags.size,
+    showStatus: false
+  })
+
+  try {
+    cli.log('Running setup and attaching to test dyno...')
+    dyno.start()
```
Comment (joshwlewis) [inline commands/ci/debug.js]: Could we log a helpful message after this resolves? Say something like "Run `sprettur setup` to setup your tests or `sprettur run` to run them.".

File: commands/ci/debug.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+function* run (context, heroku) {
+  const coupling = yield api.pipelineCoupling(heroku, context.app)
+  const pipeline = coupling.pipeline
+  const commit = yield git.readCommit('HEAD')
+  const sourceBlobUrl = yield cli.action('Preparing source', co(function* () {
+    return yield source.createSourceBlob(commit.ref, context, heroku)
+  }))
+
+  const pipelineRepository = yield api.pipelineRepository(heroku, pipeline.id)
```
Comment (joshwlewis) [inline commands/ci/debug.js]: We should check for the pipelineRepository before creating the source blob, as the pipelineRepository fetch may fail and render the uploaded blob unused and wasteful.

File: commands/ci/debug.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      organization,
+      source_blob_url: sourceBlobUrl,
+      debug: true
+    })
+
+    return yield TestRun.waitForStates(['debugging', 'errored'], run, { heroku })
+  }))
+
+  if (testRun.status === 'errored') {
+    cli.exit(1, `Test run creation failed whilst ${testRun.error_state} with message "${testRun.message}"`)
```
Comment (joshwlewis) [inline commands/ci/debug.js]: "Whilst" isn't really used in American English. I'd suggest "while" or "during" instead.

Comment (joshwlewis) [inline commands/ci/debug.js]: I don't think of "whilst" as wrong, I would just expect the Heroku voice to be American English.

File: commands/ci/debug.js (hunk 2 above)
Comment (appleton) [inline commands/ci/debug.js]: It doesn't resolve until after the dyno is killed but I guess we could do an `echo "helpful stuff"` in the command we send?

**Pull request merged**